pkg/machine/shim/host.go

```diff
@@ -557,6 +577,9 @@ func Start(mc *vmconfigs.MachineConfig, mp vmconfigs.VMProvider, dirs *machineDe
 		return errors.New(msg)
 	}
 
+	// now that the machine has transitioned into the running state, we don't need a goroutine listening for SIGINT or SIGTERM to handle state
```

Comment:
you would need to call `signal.Stop(signalChan)` first, signals/channels are hard to get right.
The reason is if you the caller close the channel then the signal code still tries to send to this channel and since the channel will be closed it will then fail to write there and panic.

Second on all early exits on errors you still leak the signal goroutine. The signal.Stop()/close(chan) call must also happen in defer that sets starting to false.

Reply:
Good catches, should be fixed. Thanks!